src/Util/setHighlight.ts

```diff
@@ -7,10 +7,13 @@ const setHighlight = (
   highlightStyle: string,
   clearHighlightRef: number,
   setClearHighlightFn: Function,
-  updateClearHandler: Function
+  updateClearHandler: Function,
+  uniqueId: number
 ) => {
-  if (!isInitial && child.ref.current) {
-    const element = child.ref.current;
+
+  const element = child.ref.current;
+
+  if (!isInitial && element && element.getAttribute('react-change-hightlight-uniqueId') == uniqueId) {
```

Comment:
use `===` not `==` for comparison, and I prefer to add the attribute name as a const in the consts file and `consts.ts` file and use it, better and easier for maintenance